calblueprint/ogsc
# Notes Filtering & Search

# Filtering and Search on Notes Table

- added sort by date (default is newest first)

-  added filtering by role and search (in contents) for the notes table


NOTE: This PR also fixes the bug with "save and send" on the edit invites page!

## TODO (if applicable)

styling is imperfect

- [x] 1. add colored button for filter dropdown
- [x] 2. figure out how to make search bar bigger
- [x] 3. color buttons

## How to Test/Use PR feature

go to a player profile -> click on notes table

1. try filtering by multiple types
2. try searching for words in the contents of the notes
3. make sure the notes only belong to that player

## PR Checklist

Does your branch (check if true):
[X] work when you run `yarn build`?
[X] successfully run `yarn:db-migrate up` without yielding any errors?
[X] successfully run `yarn prisma introspect` without yielding any errors?
[X] successfully run `yarn dev` and support all changes that your branch intends to perform?

## Related PRs

N/A

## Migrations

N/A

## Screenshots

EDIT: added some colors to the role tags!
![image](https://user-images.githubusercontent.com/65790318/115992582-48772a00-a583-11eb-94e5-31c6ff9ece1b.png)

Filtering

![image](https://user-images.githubusercontent.com/65790318/115957142-a63d4080-a4b5-11eb-861f-e1d707988851.png)


Search
![image](https://user-images.githubusercontent.com/65790318/115957188-e00e4700-a4b5-11eb-87be-c0688527adee.png)

Sort
![image](https://user-images.githubusercontent.com/65790318/116483659-5c998080-a83c-11eb-8d57-41db1dd93153.png)



CC: @sydbui
